internal/bundler/bundler.go

```diff
@@ -166,8 +167,19 @@ func parseFile(
 		ast := parser.ModuleExportsAST(log, source, parseOptions, expr)
 		results <- parseResult{source, ast, true}
 
+	case LoaderURL:
+		url := path.Base(sourcePath)
+		targetFolder := bundleOptions.AbsOutputDir
+		if targetFolder == "" {
+			targetFolder = path.Dir(bundleOptions.AbsOutputFile)
+		}
+		defer ioutil.WriteFile(path.Join(targetFolder, url), []byte(source.Contents), 0644)
```

Comment:
The bundler isn't supposed to write files itself. The bundler returns all of the files to be written to the caller. This makes the bundler easier to unit test and also makes it easy for the API to return results without modifying the file system (see for example #139).

Doing things this way also defeats future tree shaking optimizations, which may end up removing this module entirely. In that case the file should not be emitted in the output directory since it's unreferenced.

Instead of writing to the file system, the contents of the file should be stashed somewhere associated with the file, probably just on the `parseResult` or `ast`. Then the bundler should return the file to the caller, probably as an extra `BundleResult` entry. But that requires generalizing `BundleResult` to handle non-JavaScript result files.

Reply:
thank you! yeah this is a part where i'm not really sure. Do you have any plans for generalizing `BundleResult`? I could try working on it but don't want to duplicate the efforts